drivers/regulator/pmic_shell.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/drivers/regulator.h>
+#include <zephyr/drivers/regulator/consumer.h>
+
+LOG_MODULE_REGISTER(regulator_shell, CONFIG_REGULATOR_LOG_LEVEL);
+
+#define GET_REGULATOR(node) DEVICE_DT_GET(node),
+
+static const struct device *supported_pmics[] = {
+	DT_FOREACH_STATUS_OKAY(regulator_pmic, GET_REGULATOR)
+	};
+static const struct device *pmic_dev;
```

Comment:
Why not use a `<device>` argument to all `regulator` shell commands as done in other shells? There's no need to make this array at compile-time. Technically, other regulator drivers can than PMIC devices can be controlled via the exact same regulator API calls.

Reply:
Changed the commands to use a `device` argument directly